numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1886,13 +1941,22 @@ io_pack(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
 {
     PyObject *obj;
     int axis = NPY_MAXDIMS;
-    static char *kwlist[] = {"a", "axis", NULL};
+    static char *kwlist[] = {"in", "axis", "order", NULL};
+    const char * c = NULL;
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O&:pack" , kwlist,
-                                     &obj, PyArray_AxisConverter, &axis)) {
+    if (!PyArg_ParseTupleAndKeywords( args, kwds, "O|O&s:pack" , kwlist,
+                &obj, PyArray_AxisConverter, &axis, &c)) {
         return NULL;
     }
-    return pack_bits(obj, axis);
+    if (c == NULL) {
+        c = "b";
+    }
+    if (c[0] != 'l' && c[0] != 'b') {
+        PyErr_SetString(PyExc_ValueError,
+                    "'order' must begin with 'l' or 'b'");
```

Comment:
Can we match python here int `int.from_bytes` (C `longobject.c:int_to_bytes_impl`), and actually require the full string?

Could consider using the internal `_PyUnicode_EqualToASCIIId`, which I think is available in the python versions we care about.

Reply:
adopted

Reply:
I used strncmp instead, which does not require Py_LIMITED_API

Reply:
We already depend on Py_LIMITED_API, so I don't think we need to care

Reply:
one of the test runners failed when I used _PyUnicode_EqualToASCII, saying the function was not defined.